aws/data_source_aws_organizations_organizational_unit_test.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAwsOrganizationsOrganizationalUnit_basic(t *testing.T) {
+	resourceName := "aws_organizations_organizational_unit.test_ou2"
+	dataSourceName := "data.aws_organizations_organizational_unit.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsOrganizationsOrganizationalUnitConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(resourceName, "name", dataSourceName, "children.0.name"),
+					resource.TestCheckResourceAttrPair(resourceName, "id", dataSourceName, "children.0.id"),
+					resource.TestCheckResourceAttrPair(resourceName, "arn", dataSourceName, "children.0.arn"),
+				),
+				/* We ExpectNonEmptyPlan due to the explicit datasource dependency on the test_ou2 resource.
+				 * See Terraform config comments for more details. */
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+const testAccDataSourceAwsOrganizationsOrganizationalUnitConfig = `
+data "aws_organizations_organization" "test" {}
+
+resource "aws_organizations_organizational_unit" "test_ou1" {
+    name = "test_ou1"
+    parent_id = "${data.aws_organizations_organization.test.roots.0.id}"
+}
+
+# We create two OUs so the output of the data source is predictable. Otherwise,
+# we'll get a longer list of OUs if others are present in the account at the
+# root level.
+
+resource "aws_organizations_organizational_unit" "test_ou2" {
+    name = "test_ou2"
+    parent_id = "${aws_organizations_organizational_unit.test_ou1.id}"
+}
+
+# We add an explicit dependency here because if we don't, Terraform will happily
+# fetch all other existing OUs, if any, which probably don't include our test OU
+# since it won't have been created yet.
+
+data "aws_organizations_organizational_unit" "test" {
+    parent_id = "${aws_organizations_organizational_unit.test_ou1.id}"
+    depends_on = [aws_organizations_organizational_unit.test_ou2]
+}
```

Comment:
As you found, the usage of [`depends_on` in a data source configuration is problematic](https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies). This issue can be skipped by using an implicit attribute reference to the second organizational unit resource so ordering can be guaranteed without the usage of `depends_on`:

```hcl
data "aws_organizations_organizational_units" "test" {
  parent_id = aws_organizations_organizational_unit.test_ou2.parent_id
}
```